src/local-editor-service.ts

```diff
@@ -203,6 +215,27 @@ export class LocalEditorService extends EditorService {
     return autocompletion + `</${e.tagName}>$0`;
   }
 
+  private getAncestorDomModuleForElement(document: Document, element: parse5.ASTNode) {
```

Comment:
Does this handle templates inside of templates? e.g. dom-repeat and dom-if?

It's a slightly nutty idea, but it just occurred to me that if all we care about is whether the element is inside a template, we could use `isPositionInsideRange` from `ast-from-source-position`. It's a bit of a weird usage, but it should be way faster, just a few int comparisons rather than a tree walk. You'll just need to first filter by filename.